addons/nuxeo-template-rendering/nuxeo-template-rendering-core/src/main/java/org/nuxeo/template/XMLSerializer.java

```diff
@@ -20,143 +20,38 @@
 
 package org.nuxeo.template;
 
-import java.text.ParseException;
-import java.text.SimpleDateFormat;
-import java.util.ArrayList;
 import java.util.List;
 
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
-import org.dom4j.Attribute;
-import org.dom4j.Document;
 import org.dom4j.DocumentException;
-import org.dom4j.DocumentFactory;
-import org.dom4j.DocumentHelper;
-import org.dom4j.Element;
-import org.dom4j.Namespace;
-import org.dom4j.QName;
-import org.dom4j.tree.DefaultElement;
-import org.nuxeo.ecm.core.api.DocumentModel;
-import org.nuxeo.template.api.InputType;
+import org.nuxeo.runtime.api.Framework;
 import org.nuxeo.template.api.TemplateInput;
+import org.nuxeo.template.serializer.service.SerializerService;
+import org.nuxeo.template.serializer.executors.Serializer;
 
 /**
- * {@link TemplateInput} parameters are stored in the {@link DocumentModel} as a single String Property via XML
- * Serialization. This class contains the Serialization/Deserialization logic.
+ * Should use now Framework.getService(SerializerService.class).getSerializer("xml")
  *
  * @author Tiry (tdelprat@nuxeo.com)
+ * @author bjalon (bjalon@qastia.com)
  */
+@Deprecated(since = "11.1")
 public class XMLSerializer {
 
-    protected static final Log log = LogFactory.getLog(XMLSerializer.class);
+    public static Serializer xmlSerializer;
 
-    public static final String XML_NAMESPACE = "http://www.nuxeo.org/DocumentTemplate";
-
-    public static final String XML_NAMESPACE_PREFIX = "nxdt";
-
-    public static final Namespace ns = new Namespace(XML_NAMESPACE_PREFIX, XML_NAMESPACE);
-
-    public static final QName fieldsTag = DocumentFactory.getInstance().createQName("templateParams", ns);
-
-    public static final QName fieldTag = DocumentFactory.getInstance().createQName("field", ns);
-
-    public static final String DATE_FORMAT = "yyyy-MM-dd HH:mm:ss.SSS";
+    public static Serializer getXmlSerializer() {
+        if (xmlSerializer == null) {
+            xmlSerializer = Framework.getService(SerializerService.class).getSerializer("xml");
```

Comment:
Don't store the result of `Framework.getService` in a field. Do the lookup every time (same issue in other classes too)